File: backend/app/schemas/story.py
from pydantic import BaseModel, Field


class StorySummary(BaseModel):
    mission_id: str
    mission_name: str
    mission_type: str
    version: str
    world: str
    series_name: str
    mission_order: int
    characters: list[str] = Field(default_factory=list)
    summary: str
    scene_count: int = 0
    matched_character: str | None = None
    matched_scene_count: int = 0
    first_matching_chunk_id: str | None = None


class StorySceneSummary(BaseModel):
    chunk_id: str
    chunk_order: int
    scene_title: str
    location: str | None = None
    characters: list[str] = Field(default_factory=list)


class StoryDetail(StorySummary):
    story_text: str
    source_url: str
    previous_mission: str | None = None
    next_mission: str | None = None
    scenes: list[StorySceneSummary] = Field(default_factory=list)


class StoryScene(StorySceneSummary):
    mission_id: str
    mission_name: str
    content: str
    source_url: str
    previous_chunk_id: str | None = None
    next_chunk_id: str | None = None


class StoryFilters(BaseModel):
    versions: list[str] = Field(default_factory=list)
    worlds: list[str] = Field(default_factory=list)
    mission_types: list[str] = Field(default_factory=list)
    series: list[str] = Field(default_factory=list)


class StoryListResponse(BaseModel):
    items: list[StorySummary]
    total: int
    filters: StoryFilters

File: frontend/src/types/story.ts
export interface StorySummary {
  mission_id: string
  mission_name: string
  mission_type: string
  version: string
  world: string
  series_name: string
  mission_order: number
  characters: string[]
  summary: string
  scene_count: number
  matched_character?: string | null
  matched_scene_count?: number
  first_matching_chunk_id?: string | null
}

export interface StoryFilters {
  versions: string[]
  worlds: string[]
  mission_types: string[]
  series: string[]
}

export interface StorySceneSummary {
  chunk_id: string
  chunk_order: number
  scene_title: string
  location?: string | null
  characters: string[]
}

export interface StoryDetail extends StorySummary {
  story_text: string
  source_url: string
  previous_mission?: string | null
  next_mission?: string | null
  scenes: StorySceneSummary[]
}

export interface StoryScene extends StorySceneSummary {
  mission_id: string
  mission_name: string
  content: string
  source_url: string
  previous_chunk_id?: string | null
  next_chunk_id?: string | null
}
